libraries/redcore/api/hal/hal.php

```diff
@@ -2300,7 +2329,7 @@ public function apiReadTypeItem(&$primaryKeys)
 		if (!empty($operations->read->item))
 		{
 			$dataGet = $this->triggerFunction('processPostData', $this->options->get('dataGet', array()), $operations->read->item);
-			$primaryKeysFromFields = RApiHalHelper::getPrimaryKeysFromFields($operations->read->item);
+			$primaryKeysFromFields = RApiHalHelper::getFieldsArray($operations->read->item);
```

Comment:
I just switched the default of getFieldsArray to bring sense to its new name, so it will display every field by default and only if specified it will just filter primary keys